internal/mesh/internal/types/proxy_configuration.go

```diff
@@ -49,3 +53,183 @@ func MutateProxyConfiguration(res *pbresource.Resource) error {
 
 	return res.Data.MarshalFrom(&proxyCfg)
 }
+
+func ValidateProxyConfiguration(res *pbresource.Resource) error {
+	decodedProxyCfg, decodeErr := resource.Decode[*pbmesh.ProxyConfiguration](res)
+	if decodeErr != nil {
+		return resource.NewErrDataParse(decodedProxyCfg.GetData(), decodeErr)
+	}
+	proxyCfg := decodedProxyCfg.GetData()
+
+	var err error
+
+	if selErr := catalog.ValidateSelector(proxyCfg.Workloads, false); selErr != nil {
+		err = multierror.Append(err, resource.ErrInvalidField{
+			Name:    "workloads",
+			Wrapped: selErr,
+		})
+	}
+
+	if proxyCfg.GetDynamicConfig() == nil && proxyCfg.GetBootstrapConfig() == nil {
+		err = multierror.Append(err, resource.ErrInvalidFields{
+			Names:   []string{"dynamic_config", "bootstrap_config"},
+			Wrapped: errMissingProxyConfigData,
+		})
+	}
+
+	// nolint:staticcheck
+	if proxyCfg.GetOpaqueConfig() != nil {
+		err = multierror.Append(err, resource.ErrInvalidField{
+			Name:    "opaque_config",
+			Wrapped: resource.ErrUnsupported,
+		})
+	}
```

Comment:
Is there a reason to keep this field around even though it's deprecated?

Reply:
Yeah we added it for migration